src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -2973,7 +3091,7 @@ trait Types
     override def mapOver(map: TypeMap): Type = {
       val pre1 = if (pre.isInstanceOf[ClassInfoType]) pre else map(pre)
       if (pre1 eq pre) this
-      else OverloadedType(pre1, alternatives)
+      else OverloadedType(pre1, alternatives) // TODO: shouldn't we also map over the alternatives?
```

Comment:
That's tricky, because we can't just go an clone `alternatives` here, as these symbols are serving double duty of carrying an info and linking to an actual member.

For the particular case of `AsSeenFrom`, we take care to [later call `memberType`](https://github.com/scala/scala/blob/37b3892fa06cf30e0335397a2d020ea757309c37/src/compiler/scala/tools/nsc/typechecker/Infer.scala#L1421) on the eventually selected alternative.